python_modules/dagster/dagster_tests/core_tests/graph_tests/test_graph.py

```diff
@@ -78,6 +81,103 @@ def the_graph(the_in):
         the_graph.execute_in_process(input_values={"the_in": "bad_value"})
 
 
+def test_source_asset_input_value():
+    asset1 = SourceAsset("asset1")
+
+    class MyIOManager(IOManager):
+        def handle_output(self, context, obj):
+            ...
+
+        def load_input(self, context):
+            assert context.asset_key == asset1.key
+            return 5
```

Comment:
as a matter of course when doing tests like these I really like to do a pattern like this:

```
    executed = {}

    class MyIOManager(IOManager):
        def handle_output(self, context, obj):
            ...

        def load_input(self, context):
            executed["yes"] = True
            assert context.asset_key == asset1.key
            return 5
      
      # later
      
      assert executed["yes"] 
```

Just to ensure that code the framework calls back into is actually called. I've caught more than one bug this way and it proves beyond all doubt to future code readers that the rest is working properly. More than once I've found test suites that asserted against success only and it turns out they have not configured things right to call back into the right user code.

Reply:
In this case, I think the fact that `load_input` returns `5` and the op body asserts that the input value is 5 provides pretty strong guarantee. That said, updated the PR with a similar assert to what you described.